pkgs/applications/kde/cantor.nix

```diff
@@ -0,0 +1,104 @@
+{ lib
+, mkDerivation
+
+, cmake
+, extra-cmake-modules
+, makeWrapper
+, shared-mime-info
+
+, qtbase
+, qtsvg
+, qttools
+, qtwebengine
+, qtxmlpatterns
+
+, poppler
+
+, karchive
+, kcompletion
+, kconfig
+, kcoreaddons
+, kcrash
+, kdoctools
+, ki18n
+, kiconthemes
+, kio
+, knewstuff
+, kparts
+, kpty
+, ktexteditor
+, ktextwidgets
+, kxmlgui
+, syntax-highlighting
+
+, libspectre
+
+# Backends. Set to null if you want to omit from the build
+, analitza
+, julia
+, libqalculate
+, luajit
+, python3
+, R
+, sage
+, sage-with-env ? sage.with-env
+}:
+
+mkDerivation rec {
+  pname = "cantor";
+
+  nativeBuildInputs = [
+    cmake
+    extra-cmake-modules
+    makeWrapper
+    shared-mime-info
+    qttools
+  ];
+
+  buildInputs = [
+    qtbase
+    qtsvg
+    qtwebengine
+    qtxmlpatterns
+
+    poppler
+
+    karchive
+    kcompletion
+    kconfig
+    kcoreaddons
+    kcrash
+    kdoctools
+    ki18n
+    kiconthemes
+    kio
+    knewstuff
+    kparts
+    kpty
+    ktexteditor
+    ktextwidgets
+    kxmlgui
+    syntax-highlighting
+
+    libspectre
+  ]
+  # backends
+  ++ lib.optional (analitza != null) analitza
+  ++ lib.optional (julia != null) julia
+  ++ lib.optional (libqalculate != null) libqalculate
+  ++ lib.optional (luajit != null) luajit
+  ++ lib.optional (python3 != null) python3
+  ++ lib.optional (R != null) R
+  ++ lib.optional (sage-with-env != null) sage-with-env
+  ;
+
+  qtWrapperArgs = [
+    "--prefix PATH : ${placeholder "out"}/bin"
+  ];
+
+  postInstall = ''
+    wrapProgram $out/share/cantor/sagebackend/cantor-execsage \
+      --prefix PATH : ${sage.with-env}/bin
```

Comment:
I am not sure if I should use wrapProgram or if I should just edit the file to have an absolute path to sage-ipython. The file's contents are

```
#!/nix/store/zlf0f88vj30sc7567b80l52d19pbdmy2-bash-5.2-p15/bin/bash

export SAGE_ROOT="$($1 --root)"

# Sage 9.2 stop source sage-env-config automatically, so do it manually
if [[ -f $SAGE_ROOT"/bin/sage-env" ]]; then
    if [[ -f $SAGE_ROOT"/bin/sage-env-config" ]]; then
        source "$SAGE_ROOT/bin/sage-env-config" >&2
    fi
    . "$SAGE_ROOT/bin/sage-env" >&2
elif [[ -f $SAGE_ROOT"/local/bin/sage-env" ]]; then
    if [[ -f $SAGE_ROOT"/local/bin/sage-env-config" ]]; then
        source "$SAGE_ROOT/local/bin/sage-env-config" >&2
    fi
    . "$SAGE_ROOT/local/bin/sage-env" >&2
elif [[ -f $SAGE_ROOT"/src/bin/sage-env" ]]; then
    if [[ -f $SAGE_ROOT"/src/bin/sage-env-config" ]]; then
        source "$SAGE_ROOT/src/bin/sage-env-config" >&2
    fi
    . "$SAGE_ROOT/src/bin/sage-env" >&2
fi

exec sage-ipython -i --simple-prompt
```